src/result_processor.c

```diff
@@ -786,8 +790,12 @@ static SearchResult *GetNextResult(RPSafeLoader *self) {
 static int rpSafeLoaderNext_Accumulate(ResultProcessor *rp, SearchResult *res);  // Forward declaration
 
 static int rpSafeLoader_ResetAndReturnLastCode(RPSafeLoader *self) {
-
-  self->base_loader.base.Next = rpSafeLoaderNext_Accumulate; // Reset the next function, in case we are in cursor mode
+  // Reset the next function, in case we are in cursor mode
+  if (self->becomePlainLoader) {
+    self->base_loader.base.Next = rploaderNext;
```

Comment:
Why not covered?

Reply:
It’s actually not possible today, because the loader knows exactly how many documents it has to load, and we never have buffered results at this point. 
This might change in the future. There are some edge cases that might benefit buffering extra results